src/main.rs

```diff
@@ -28,19 +37,45 @@ struct CompilerOptions {
     /// Set the maximum stack size for the current running instance.
     #[clap(short, long, default_value = "10000")]
     stack_size: u32,
+
+    /// Run the compiler in debug mode
+    #[clap(short, long)]
+    debug: bool,
 }
 
 fn main() {
     let opts: CompilerOptions = CompilerOptions::parse();
 
-    println!("Stack_size is {}", opts.stack_size);
+    // print recieved cmdargs, if debug is specified
+    if opts.debug {
```

Comment:
Maybe we wanna use the `log` create in conjunction with some logging lib like `pretty_env_logger`? This way we can avoid always writing `if opts.debug`, among other things.

Reply:
Ah well, I didn't know that was a thing. Ideally, we want to set up a logger for each sub-system to enable debug in specific parts of the compiler for a much better dev experience.

Reply:
Yeah, `pretty_env_logger` supports stuff like that. https://docs.rs/env_logger/0.8.3/env_logger/